src/app/routes/connector-ui/contract-agreement-page/contract-agreement-page/contract-agreement-page.service.ts

```diff
@@ -43,30 +46,54 @@ export class ContractAgreementPageService {
   }
 
   private fetchData(): Observable<Fetched<ContractAgreementPageData>> {
-    return this.edcApiService.getContractAgreementPage().pipe(
-      Fetched.wrap({failureMessage: 'Failed fetching contract definitions'}),
-      Fetched.map((contractAgreementPage) =>
-        this.buildContractAgreementPageData(contractAgreementPage),
+    return combineLatest([
+      this.edcApiService.getContractAgreementPage(),
+      this.fetchLimits(),
+    ]).pipe(
+      map(([contractAgreementPage, connectorLimits]) =>
+        this.buildContractAgreementPageData(
+          contractAgreementPage,
+          connectorLimits,
+        ),
       ),
+      Fetched.wrap({failureMessage: 'Failed fetching Contract Agreement Page'}),
     );
   }
 
   private buildContractAgreementPageData(
     contractAgreementPage: ContractAgreementPage,
+    connectorLimits: ConnectorLimits | null,
   ): ContractAgreementPageData {
     let contractAgreements = this.mapContractAgreements(
       contractAgreementPage.contractAgreements,
     );
+
+    let consumingContractAgreements = contractAgreements.filter(
+      (it) => it.direction === 'CONSUMING',
+    );
+    let providingContractAgreements = contractAgreements.filter(
+      (it) => it.direction === 'PROVIDING',
+    );
+
+    let isConsumingLimitsEnforced =
+      connectorLimits?.maxActiveConsumingContractAgreements != null &&
+      connectorLimits.maxActiveConsumingContractAgreements >= 0;
+    if (isConsumingLimitsEnforced) {
+      consumingContractAgreements =
+        this.contractAgreementCardMappedService.setStatus(
+          connectorLimits?.maxActiveConsumingContractAgreements!,
+          consumingContractAgreements,
+        );
+    }
```

Comment:
it is better if this is included in the contractAgreementCardMappedService.buildContractAgreementCardMapped` call.

otherwise we have two calls "mutating" / building this ContractAgreementCardMapped, which can be confusing.

unless we have a good reason, we should build immutable objects, and build them at once. so we have only one place to look when it comes to checking where data comes from